src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -138,6 +147,14 @@ class SharedLinkSection extends React.Component<Props, State> {
             }
         }
 
+        if (
+            prevProps.sharedLink.permissionLevel !== '' &&
```

Comment:
Added an empty string check here because by default permissionLevel is always empty string when a sharedLink is first instantiated in the parent app. We don't want to prematurely set `isPermissionElevatedToEdit`.